tor/core/validation.py

```diff
@@ -73,11 +73,16 @@ def verified_posted_transcript(post, config):
                 return False
 
     # get source link, check all comments, look for root level comment
-    # by the author of the post. Return True if found, False if not.
+    # by the author of the post and verify that the key is in their post.
+    # Return True if found, False if not.
     linked_resource = config.r.submission(top_parent.id_from_url(top_parent.url))
     # get rid of the "See More Comments" crap
     linked_resource.comments.replace_more(limit=0)
     for top_level_comment in linked_resource.comments.list():
-        if post.author == top_level_comment.author:
-            return True
+        if _author_check(post, top_level_comment):
+            # we can do an 'and' statement above, but that would require
+            # running the header check on every comment as well as the author
+            # check and we want to save cycles wherever we can.
```

Comment:
That's not true, is it? When using `and`, the right side is only even evaluated if the left side is true. So using `and` would be the same as having nested `if`s.

Reply:
![screenshot 2017-09-06 09 54 08](https://user-images.githubusercontent.com/5179553/30115685-759a8d48-92e9-11e7-86ef-0005b8581bf5.png)

You're totally right. Oops.